services-api/src/main/java/io/scalecube/services/registry/api/ServiceRegistry.java

```diff
@@ -23,7 +23,9 @@
 
   ServiceEndpoint unregisterService(String endpointId);
 
-  Flux<RegistryEvent> listen();
+  Flux<ReferenceRegistryEvent> listenReferenceEvents();
```

Comment:
i dont like this contract its bloated to have this kind in the interface

maybe we can provide more clean interface?

Reply:
something like:

```
listen(RegistryEvent::reference)

listen(RegistryEvent::endpoint)
```